metaflow/client/core.py

```diff
@@ -1123,6 +1123,139 @@ def _iter_filter(self, x):
         # exclude private data artifacts
         return x.id[0] != "_"
 
+    def _iter_matching_tasks(self, steps, pattern):
+        """
+        Yield tasks from specified steps matching a foreach path pattern.
+
+        Parameters
+        ----------
+        steps : List[str]
+            List of step names to search for tasks
+        pattern : str
+            Regex pattern to match foreach-indices metadata
+
+        Returns
+        -------
+        Iterator[Task]
+            Tasks matching the foreach path pattern
+        """
+        flow_id, run_id, _, _ = self.path_components
+
+        for step in steps:
+            task_pathspecs = self._metaflow.metadata.filter_tasks_by_metadata(
+                flow_id, run_id, step, "foreach-execution-path", pattern
+            )
+            for task_pathspec in task_pathspecs:
+                yield Task(pathspec=task_pathspec, _namespace_check=False)
+
+    @property
+    def parent_tasks(self) -> List["Task"]:
+        """
+        Returns a list of all parent tasks of the current task for the latest successful
+        attempt.
+
+        Returns
+        -------
+        List["Task"]
+            List of all parent tasks of the current task.
+        """
+        flow_id, run_id, _, _ = self.path_components
+        steps = self.parent.parent_steps
+        if not steps:
+            return
+
+        current_path = self.metadata_dict.get("foreach-execution-path", "")
```

Comment:
nit - just `self.metadata_dict.get("foreach-execution-path")` - just to guard against an eventuality where foreach-execution-path set to empty starts having a meaning.